pkg/api/customization/oci/listers.go

```diff
@@ -0,0 +1,238 @@
+package oci
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/rancher/norman/httperror"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/oracle/oci-go-sdk/common"
+	"github.com/oracle/oci-go-sdk/containerengine"
+	"github.com/oracle/oci-go-sdk/core"
+	"github.com/oracle/oci-go-sdk/identity"
+
+	"encoding/json"
+)
+
+func processVcns(provider common.ConfigurationProvider, compartment string) ([]byte, httperror.ErrorCode, error) {
```

Comment:
All the helpers in here should be updated to return an int in place of httperror.ErrorCode, then in the error handling above that can be passed strait into the helpers.